pallets/tx_fee/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::weights::{DispatchInfo, GetDispatchInfo};
+use pallet_transaction_payment::{FeeDetails, OnChargeTransaction};
+use sp_runtime::{traits::Dispatchable, FixedPointOperand};
+
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+
+// Type aliases used for interaction with `OnChargeTransaction`.
+pub(crate) type OnChargeTransactionOf<T> =
+	<T as pallet_transaction_payment::Config>::OnChargeTransaction;
+
+// Balance type alias.
+pub(crate) type BalanceOf<T> = <OnChargeTransactionOf<T> as OnChargeTransaction<T>>::Balance;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_transaction_payment::Config {}
+
```

Comment:
Extend transaction payment config for this pallet here we can add customization specific to MRC